app/src/main/java/emu/skyline/provider/DocumentsProvider.kt

```diff
@@ -0,0 +1,269 @@
+/*
+ * SPDX-License-Identifier: MPL-2.0
+ * Copyright © 2022 Skyline Team and Contributors (https://github.com/skyline-emu/)
+ */
+
+package emu.skyline.provider
+
+import android.database.Cursor
+import android.database.MatrixCursor
+import android.os.CancellationSignal
+import android.os.ParcelFileDescriptor
+import android.provider.DocumentsContract
+import android.provider.DocumentsProvider
+import android.webkit.MimeTypeMap
+import emu.skyline.R
+import emu.skyline.SkylineApplication
+import emu.skyline.getPublicFilesDir
+import java.io.*
+
+class DocumentsProvider : DocumentsProvider() {
+
+    private val baseDirectory = File(SkylineApplication.instance.applicationContext.getPublicFilesDir().canonicalPath)
```

Comment:
In case you didn't already know, `instance` is equivalent to `applicationContext` (much like you can use an activity as its context object, this calls application.applicationContext)

Reply:
Oh my bad I didn't know that i'll change that right now with my other fixes